### PR TITLE
Whitelisting changes

### PR DESCRIPTION
**Pull request by PromoFaux**
Right, this is a lot more robust than my previous pull.
@jacobsalmela @dschaper @Mcat12 take a look, and see what you think.

(I even did it on a different branch this time!)

### CODE REVIEW

**Comment by Mcat12** [comment]
While you did do your work on a different branch (good job with that!), you are still requesting to merge into the main branch. Also, it may be better to change the name of your branch to "whitelist", as that is the name of the branch on this repo where whitelist changes are occurring. If you do that, you should change where you're merging into from "master" to "whitelist". 

That's just from glancing at the request. I'll look into the actual changes later, when I have a bit more time. :D

File: advanced/Scripts/blacklist.sh [hunk 1]
```diff
@@ -0,0 +1,180 @@
...
+	
+	if [ -z "$validDomain" ]; then
+		echo $1 is not a valid argument or domain name
+	else	  
+	  domList=("${domList[@]}" $validDomain)
+	fi
+}
+
+function PopBlacklistFile(){
+	#check whitelist file exists, and if not, create it
```
Comment (Mcat12) [inline advanced/Scripts/blacklist.sh]: Looks like you forgot to change whitelist to blacklist in the comment

Comment (PromoFaux) [inline advanced/Scripts/blacklist.sh]: heh, semantics. that's easily fixed.

**Comment by PromoFaux** [comment]
>Also, it may be better to change the name of your branch to "whitelist", as that is the name of the branch on this repo where whitelist changes are occurring. If you do that, you should change where you're merging into from "master" to "whitelist".

Sure thing, I can look into that when I get home. I'm still fairly new to git, just finding my legs!

**Comment by PromoFaux** [comment]
see #174

**Pull request closed without merging**